### DISCUSSION TITLE
ImageSource in C#?

### DISCUSSION THREAD

**Discussion by eamonnalphin** [AathifMahir/MauiIcons#90] [Q&A] [answered]
Is it possible to use MauiIcons as an imageSource, in a swipeView or A ShellContent Icon? 

I can't seem to figure out how to do this in C#.  I can see it's possible to use MauiIcons as an image source in XAML, but I can't figure out how convert that into C#.

**Comment by AathifMahir** [comment]
> Is it possible to use MauiIcons as an imageSource, in a swipeView or A ShellContent Icon? 
> 
> I can't seem to figure out how to do this in C#.  I can see it's possible to use MauiIcons as an image source in XAML, but I can't figure out how convert that into C#. 
> 
> 

Of course, we can use MauiIcons as image source or fontimage source, but can you provide a sample of your use case. Therefore I can provide you a optimal way of making the mauiicons as image source

**Comment by eamonnalphin** [comment]
Oh wow that was fast...um....

I've got a ShellContent view, built into Maui, which has an Icon property which expects an ImageSource, and the SwipeItem class which has an IconImageSource property, expecting an ImageSource, and I don't know how to get the ImageSource out of a MauiIcon: 

```
pubic class LogView : ShellContent{

public LogView(){
    Title = "Log",
    Icon = "What Do I put here?"

    var swipeItem = new SwipeItem {
        IconImageSource="What goes here?"
    }
    var mySwipeView = new SwipeView();
    mySwipeView.RightItems = new SwipeItems{swipeItem};
    //then add to view
}

}

**Reply by AathifMahir** [reply]
> Oh wow that was fast...um....
> 
> I've got a ShellContent view, built into Maui, which has an Icon property which expects an ImageSource, and the SwipeItem class which has an IconImageSource property, expecting an ImageSource, and I don't know how to get the ImageSource out of a MauiIcon: 
> 
> ```
> pubic class LogView : ShellContent{
> 
> public LogView(){
>     Title = "Log",
>     Icon = "What Do I put here?"
> 
>     var swipeItem = new SwipeItem {
>         IconImageSource="What goes here?"
>     }
>     var mySwipeView = new SwipeView();
>     mySwipeView.RightItems = new SwipeItems{swipeItem};
>     //then add to view
> }
> 
> }

Sure, got it. Let me try the MauiIcons Extension and also casting the built in control and get back to you with a an optimal solution

**Reply by eamonnalphin** [reply]
I appreciate it, thanks! :)

**Comment by AathifMahir** [comment, marked as answer]
@eamonnalphin I have release v2.1.5 of MauiIcons which Contains new Extension for your Use Case, As you can see in the Below Code, I Have Used MauiIcons.Material Module to Demonstrate the New Extension, You can use this extension on All of MauiIcons Modules

```csharp
using MauiIcons.Core;
using MauiIcons.Material;

public class LogView : ShellContent
{
  public LogView()
  {
    Title = "Log";
    Icon = MaterialIcons.Favorite.ToImageSource();

    var swipeItem = new SwipeItem {
        IconImageSource= MaterialIcons.Favorite.ToImageSource();
    };
    var mySwipeView = new SwipeView();
    mySwipeView.RightItems = new SwipeItems{swipeItem};
  }
}
```

**Comment by AathifMahir** [comment]
I'll mark this answered and close this discussion, feel free to comment on this if you have any questions

**Reply by eamonnalphin** [reply]
Thanks so much for the quick fix! I'll give it a try as soon as I get a chance!
